test/rules/src/https_everywhere_checker/http_client.py

```diff
@@ -379,14 +377,12 @@ def fetchHtml(self, url):
 			if httpCode == 0:
 				raise HTTPFetcherError("Pycurl fetch failed for '%s'" % newUrl)
 			elif httpCode in (301, 302, 303, 307):
-				# Parse out the headers and extract location, case-insensitively.
-				# If there are multiple location headers, pick the last one.
-				headers = dict()
-				for k, v in self._headerRe.findall(headerStr):
-					headers[k.lower()] = v
-				location = headers.get('location')
-				if not location:
-					raise HTTPFetcherError("Redirect for '%s' missing Location" % newUrl)
+				location = None
+				for piece in headerStr.lower().split('\n'):
+					if piece.startswith('location:'):
+						location = ':'.join(piece.split(':')[1:]).strip()
```

Comment:
Right again, in fact that comes up in the https://itunes.com example because the `Location` there is http://www.apple.com/itunes/?cid=OAS-US-DOMAINS-itunes.com . I'll fix it tomorrow.